[HDR] Use a real gain-map image for layout tests

shallawa opened the pull request:
#### e51de5df0de45a86e5416e2792231db43a52f3bc
<pre>
[HDR] Use a real gain-map image for layout tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=297422">https://bugs.webkit.org/show_bug.cgi?id=297422</a>
<a href="https://rdar.apple.com/158354774">rdar://158354774</a>

Reviewed by Simon Fraser.

This will test decoding and displaying HDR images if the underlying system
frameworks support HDR images.

* LayoutTests/fast/images/hdr-background-image-expected.html:
* LayoutTests/fast/images/hdr-background-image.html:
* LayoutTests/fast/images/hdr-basic-image-dynamic-range-limit-expected.html:
* LayoutTests/fast/images/hdr-basic-image-dynamic-range-limit.html:
* LayoutTests/fast/images/hdr-basic-image-expected.html:
* LayoutTests/fast/images/hdr-basic-image.html:
* LayoutTests/fast/images/hdr-border-image-expected.html:
* LayoutTests/fast/images/hdr-border-image.html:
* LayoutTests/fast/images/hdr-image-layer-sdr-dynamic-range-limit-expected.html:
* LayoutTests/fast/images/hdr-image-layer-sdr-dynamic-range-limit.html:
* LayoutTests/fast/images/hdr-pseudo-before-image-expected.html:
* LayoutTests/fast/images/hdr-pseudo-before-image.html:
* LayoutTests/fast/images/hdr-svg-inline-image-expected.html:
* LayoutTests/fast/images/hdr-svg-inline-image.html:
* LayoutTests/fast/images/resources/gainmap-red-green-1920x1920.jpg: Added.
* Source/WebCore/platform/graphics/BitmapImage.cpp:
(WebCore::BitmapImage::draw):

Canonical link: <a href="https://commits.webkit.org/298828@main">https://commits.webkit.org/298828@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/37689166b922a4e90f1340d45e7a7de997f974bd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116789 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36453 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27026 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122865 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67383 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/75f4a286-432f-47a4-bfcd-380baeb4b8e3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118678 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37151 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45054 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88689 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/43103 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1db5c0ad-9b01-4cd8-857d-9e1f2518f3e3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119738 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29620 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104767 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69159 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28683 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22872 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66530 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98999 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23028 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125999 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43688 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32810 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97358 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44052 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100969 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97156 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24736 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42481 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20420 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39687 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43574 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49170 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43041 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46380 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44746 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on LayoutTests/fast/images/resources/gainmap-red-green-100x100.jpg:
This image is 1920x1920, not 100x100

shallawa committed:
[HDR] Use a real gain-map image for layout tests
https://bugs.webkit.org/show_bug.cgi?id=297422
rdar://158354774

Reviewed by Simon Fraser.

This will test decoding and displaying HDR images if the underlying system
frameworks support HDR images.

* LayoutTests/fast/images/hdr-background-image-expected.html:
* LayoutTests/fast/images/hdr-background-image.html:
* LayoutTests/fast/images/hdr-basic-image-dynamic-range-limit-expected.html:
* LayoutTests/fast/images/hdr-basic-image-dynamic-range-limit.html:
* LayoutTests/fast/images/hdr-basic-image-expected.html:
* LayoutTests/fast/images/hdr-basic-image.html:
* LayoutTests/fast/images/hdr-border-image-expected.html:
* LayoutTests/fast/images/hdr-border-image.html:
* LayoutTests/fast/images/hdr-image-layer-sdr-dynamic-range-limit-expected.html:
* LayoutTests/fast/images/hdr-image-layer-sdr-dynamic-range-limit.html:
* LayoutTests/fast/images/hdr-pseudo-before-image-expected.html:
* LayoutTests/fast/images/hdr-pseudo-before-image.html:
* LayoutTests/fast/images/hdr-svg-inline-image-expected.html:
* LayoutTests/fast/images/hdr-svg-inline-image.html:
* LayoutTests/fast/images/resources/gainmap-red-green-1920x1920.jpg: Added.
* Source/WebCore/platform/graphics/BitmapImage.cpp:
(WebCore::BitmapImage::draw):

Canonical link: https://commits.webkit.org/298828@main
